core/trino-main/src/main/java/io/trino/sql/planner/EqualityInference.java

```diff
@@ -44,6 +44,7 @@
 import static io.trino.sql.planner.DeterminismEvaluator.isDeterministic;
```

Comment:
Could you add a commit description saying what is fixed and what was the problem?

Reply:
The issue to be fixed is described in this PR. But sure I can add similar description to the commit, once I get your review on the approach and possible review comments :) I will add the comment/description altogether.